src/voice_registration.js

```diff
@@ -417,11 +467,42 @@ go.app = function() {
                     self.im, name, lang, speech_option),
                 next: function(content) {
                     var period_date = content+"-"+monthNum+"-"+year;
-                    if (!go.utils.is_valid_date(period_date, 'DD-MM-YYYY')) {
-                        return 'state_retry_last_period_day';
-                    } else {
-                        self.im.user.set_answer('last_period_date', period_date);
-                        return 'state_msg_language';
+                    self.im.user.set_answer('last_period_date', period_date);
+
+                    return 'state_validate_date';
+                }
+            });
+        });
+
+        // to validate overall date
+        self.add('state_validate_date', function(name) {
+            var dateToValidate = self.im.user.answers.last_period_date ||
+                                 self.im.user.answers.baby_birth_date;
+
+            if (go.utils.is_valid_date(dateToValidate, 'DD-MM-YYYY')) {
+                return self.states.create('state_msg_language');
+            } else {
+                return self.states.create('state_invalid_date', {date: dateToValidate});
+            }
+        });
+
+        self.add('state_invalid_date', function(name, opts) {
+            var speech_option = 1;
+            return new ChoiceState(name, {
+                question:
+                    $('{{ date }} is not a real date. Please try again.'
+                    ).context({date: opts.date}),
```

Comment:
Since this is voice we will not be able to read each different date, so just replace `{{date}}` with `The date you entered` and remove passing the date through opts.